### ISSUE TITLE
Linux version?

### ISSUE THREAD

**Issue by ViBE-HU** [x360ce/x360ce#464]
Since I'm using this application with my no name controller I'm very satisfied. Most of my games recognise it as a genuine XBOX controller on Windows. But on Linux (Ubuntu 16.04) I did not find the same solution and other methods are pretty useless and the controller is just don't work. Could you make a Linux version please? Is it possible?

**Comment by Nucleoprotein** [comment]
Not possible, Linux apps does not use Xinput, and for Wine is also not possible because translation will screw-up hooking.
XBOX Controllers works fine on Linux - you can use xpad kernel driver or xboxdrv: https://github.com/xboxdrv/xboxdrv (for ubuntu: https://github.com/raelgc/ubuntu_xboxdrv)

**Comment by Enverex** [comment]
Just thought it's worth noting for anyone else finding this (and @Nucleoprotein) that x360ce works fine in Wine when using the xboxdrv Linux driver (with mimic-xpad on).

**Comment by chewi** [comment]
@Enverex …after you install dotnet, which sometimes breaks the game you're actually trying to play. Even then, I never managed to get the GUI to work properly.

**Comment by Enverex** [comment]
You have to run the exe without any of the DLLs in place with DotNet4.0 installed. But that's only if you need to use the exe, which, after you've set up once, you don't seem to need.

I have an .ini file and the xinput1_3.dll file which I can drop into any xinput game and all my controllers work perfectly - xboxdrv turns everything into a common fake 360 controller and then x360ce uses those fake controllers to fake xinput.

I've actually only run into one game that it doesn't work for (Rayman Legends, game crashes if I use x360ce's xinput dll which is odd).

Basically, if you're using xboxdrv on Linux, you shouldn't actually need the program itself, only these two premade files.

**Comment by chewi** [comment]
@Enverex I wasn't sure if you only needed dotnet for the GUI and I wasn't able to find out since I couldn't create the .ini file. Please could you paste yours somewhere? I'd be very grateful.

**Comment by Enverex** [comment]
Sure, here's my ini - https://emerald.xnode.org/x360ce.ini

Basically disable the "(event)" version in Wine just leaving the "(js)" version. The way it works is...

```
(Any controllers) > xboxdrv > x360ce > xinput
```

Because xboxdrv always reports the mapped controllers as being the same 4 XBox controllers, x360ce will always "just work" as it's mapped to those 4 virtual controllers.

**Comment by ColinKinloch** [comment]
Here's that file as a gist in case of link rot:
https://gist.githubusercontent.com/ColinKinloch/e3359faee09647b188ecda6e839dc505/raw/881f302304882024cebd6fc913c2bc4af72aa2ae/x360ce.ini
